src/rlookup.c

```diff
@@ -595,7 +600,7 @@ static int RLookup_JSON_GetAll(RLookup *it, RLookupRow *dst, RLookupLoadOptions
   //RedisModuleString *krstr =
   //    RedisModule_CreateString(ctx, options->dmd->keyPtr, sdslen(options->dmd->keyPtr));
   // TODO: check error
-  RedisJSONKey jsonKey = japi->openKeyFromStr(ctx, options->dmd->keyPtr);
+  RedisJSONKey jsonKey = japi ? japi->openKeyFromStr(ctx, options->dmd->keyPtr) : NULL;
```

Comment:
I think it's better to explicitly ask for `!japi` here and to bail out. Going through `done` looks a little entangled to me.

Reply:
One more thing to consider is that this entangled situation is actually rather unexpected, and this is just additional precaution, leading to the "branch taken" approach (positive/sunny-day scenario).
Does this consideration make a difference?